### ISSUE TITLE
Add ability to suppress emscripten warnings

### ISSUE THREAD

**Issue by ricochet** [kripken/emscripten#4473]
Some emscripten settings will always cause warnings to be emitted for every invocation of emcc (typically from emcc.py). For a large project like mine, this can mean 10,000+ warnings. Most of them are "you better know what you are doing" types of warnings and can safely be suppressed. 

For example, the separate-asm switch produces the following warning:

> WARNING:root:--separate-asm works best when compiling to HTML. otherwise, you must yourself load the '.asm.js' file that is emitted separately, and must do so before loading the main '.js` file

Perhaps a simple fix would be to add new command line arguments for suppressing emscripten warnings or support warning exclusions in the config. My first thought was to add "-Wno-*" switches but that could get messy unless these were stripped from the command line settings sent to the compiler.

> When an unrecognized warning option is requested (e.g., -Wunknown-warning), GCC emits a diagnostic stating that the option is not recognized. However, if the -Wno- form is used, the behavior is slightly different: no diagnostic is produced for -Wno-unknown-warning unless other diagnostics are being produced. This allows the use of new -Wno- options with old compilers, but if something goes wrong, the compiler warns that an unrecognized option is present.
> http://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

**Comment by kripken** [comment]
Sounds like a good idea. `emcc.py` could consume the `-Wno-` things specific to it, and hide those warnings while also not passing those on to clang. We'd have a list of which are emcc's, or something like that.

It's possible some of those warnings aren't shown in `emcc.py`, I'm not sure. If so we might want some code in `tools/shared.py` to manage this (and then `emcc.py` and other relevant locations could call it).

**Comment by dschuff** [comment]
Another option might be to put the suppressions in the config file. Then there would be no ambiguity between the custom emscripten warnings and the warnings from the underlying compiler. Would be easier for build systems too.

**Comment by ricochet** [comment]
From a design perspective, do you view emcc as a replacement for gcc/clang or is it a higher level sdk piece that wraps the compiler? The latter works well with the config implementation and the former would match `-Wno-`. 

Given your preference, I can probably whip something up next week.

**Comment by kripken** [comment]
Hmm, mostly as a drop-in replacement I guess, but it also does higher-level things.

I see that we already do support disabling some of our new warnings, here: https://github.com/kripken/emscripten/blob/incoming/emcc.py#L465 Continuing in that direction seems reasonable. Might be nice to refactor that code into an object in `tools/shared.py` (`WarningManager` perhaps?) and just call it from `emcc.py`, in particular since I think well need to call it (to check if to show a warning or not) from other places too.

**Comment by juj** [comment]
@ricochet: Do you have a specific list of what kind of warnings you are running into?

**Comment by ricochet** [comment]
The ones that I care about are from `separate-asm` and `ALLOW_MEMORY_GROWTH`. 

```python
logging.warning('not all asm.js optimizations are possible with ALLOW_MEMORY_GROWTH, disabling those')
```

When experimenting with `SPLIT_MEMORY`, I ran into a similar warning.

The last time I ran a build with `--profiling` I saw a ton of "function xxx not implemented" warnings. My app ran without issue, but I need to go back and sanity check what I was doing at the time. That's unexpected, right?
